db/migrate/20240818113758_remove_all_but_most_recent_user_password.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+class RemoveAllButMostRecentUserPassword < ActiveRecord::Migration[7.1]
+  def up
+    execute <<~SQL.squish
+      DELETE FROM user_passwords
+      WHERE id NOT IN (
+        SELECT DISTINCT ON (user_id) id
+        FROM user_passwords
+        ORDER BY user_id, password_expired_at IS NULL DESC, password_expired_at DESC
```

Comment:
`NULLS FIRST` / `NULLS LAST` might be nice here?
```suggestion
        ORDER BY user_id, password_expired_at DESC NULLS FIRST
```